nameko/containers.py

```diff
@@ -436,44 +448,43 @@ def _run_worker(self, worker_ctx, handle_result):
                 del exc_info
 
                 self.dependencies.all.worker_teardown(worker_ctx)
-                # release?
 
-    def _kill_active_threads(self):
-        """ Kill all managed threads that were not marked as "protected" when
-        they were spawned.
+    def _kill_worker_threads(self):
+        """ Kill any currently executing worker threads.
 
-        This set will include all worker threads generated by
-        :meth:`ServiceContainer.spawn_worker`.
-
-        See :meth:`ServiceContainer.spawn_managed_thread`
+        See :meth:`ServiceContainer.spawn_worker`
         """
-        num_active_threads = len(self._active_threads)
+        num_workers = len(self._worker_threads)
 
-        if num_active_threads:
-            _log.warning('killing %s active thread(s)', num_active_threads)
-            for gt, extension in list(self._active_threads.items()):
-                if extension is not MANAGED_THREAD:
-                    _log.warning('killing active thread for %s', extension)
+        if num_workers:
+            _log.warning('killing %s active workers(s)', num_workers)
+            for worker_ctx, gt in self._worker_threads.copy().items():
```

Comment:
Is it? In Py3 `dict.items()` will give you back a `dict_items` object, so if you `copy()` you're also iterating over a `dict_items` (which is a generator, I think) rather than reading them all into a list first.